R/board_url.R

```diff
@@ -141,12 +205,100 @@ pin_store.pins_board_url <- function(board, name, paths, metadata,
   abort_board_read_only("board_url")
 }
 
+#' @export
+pin_version_delete.pins_board_url <- function(board, name, version, ...) {
+  abort_board_read_only("board_url")
+}
+
 #' @export
 write_board_manifest_yaml.pins_board_url <- function(board, manifest, ...) {
   abort_board_read_only("board_url")
 }
 
 # Helpers ------------------------------------------------------------------
+
+board_url_error_message <- c(
+  "{.var urls} must resolve to either:",
+  "*" = "an unnamed character scalar, i.e. a single URL",
+  "*" = "a named character vector",
+  "*" = "a named list, where all elements are character scalars or vectors"
+)
+
+get_url_format <- function(urls) {
+  if (is_scalar_character(urls) && !is_named(urls)) {
+    return ("pins_yaml")
+  }
+  if (is_list(urls) && is_named(urls) && all(map_lgl(urls, is_character))) {
+    return ("manifest_content")
+  }
+  if (is.character(urls) && is_named(urls)) {
+    return ("vector_of_urls")
+  }
+  cli::cli_abort(
+    board_url_error_message,
+    class = "pins_error_board_url_argument",
+    urls = urls
+  )
```

Comment:
Stylistically, I think this makes it clearer that the job of this function is to return one of three strings, or die trying: 

```suggestion
  if (is_scalar_character(urls) && !is_named(urls)) {
    "pins_yaml"
  } else if (is_list(urls) && is_named(urls) && all(map_lgl(urls, is_character))) {
    "manifest_content"
  } else if (is.character(urls) && is_named(urls)) {
    "vector_of_urls"
  } else {
    cli::cli_abort(
      board_url_error_message,
      class = "pins_error_board_url_argument",
      urls = urls
    )
  }
```

Reply:
I think you can now inline `board_url_error_message `?